auth/tasks.py

```diff
@@ -50,51 +64,77 @@ def update_group_and_perms(department, created):
         if created:
             assign_model_perms_for_department(department)
 
+    def remove_related_department_group(super_department, childs):
+        # 将当前department的所有叶子结点返回
+        child_departments = Department.child_departments.all()
+        if child_departments:
+            for child_department in child_departments:
+                childs.extend(
+                    remove_related_department_group(child_department, childs))
+        else:
+            childs.extend([super_department])
+        return childs
+    try:
+        for raw_department in raw_departments:
+            department, created = Department.objects.get_or_create(
+                raw_department_id=raw_department.dwid,
+                defaults={'name': raw_department.dwmc})
+            update_group_and_perms(department, created)
+            updated = False
+            # 同步隶属单位
+            if (department.super_department is None or
+                    department.super_department.raw_department_id !=
+                    raw_department.lsdw):
+                super_department_name = DepartmentInformation.objects.get(
+                    dwid=raw_department.lsdw).dwmc
+                super_department, created = Department.objects.get_or_create(
+                    raw_department_id=raw_department.lsdw,
+                    defaults={'name': super_department_name}
+                )
+                update_group_and_perms(super_department, created)
+                # 将老师从原有group中删除
+                if department.super_department:
+                    child_departments = remove_related_department_group(
+                        department.super_department, [])
+                    for department in child_departments:
+                        teachers = User.objects.filter(department=department)
+                        for teacher in teachers:
```

Comment:
As you argued offline yesterday, updating users in an update function for departments is context-related, this function shouldn't know any thing about updating users.